File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-msg-5-4/ServiceNotification.cpp
// ServiceNotification.cpp 
#include "ServiceNotification.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ServiceNotification::ServiceNotification(TiXmlNode* xmlNode)
: NotificationMessage(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //serviceNameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* serviceNameNode = xmlNode->FirstChildElement("serviceName");

   if(serviceNameNode){serviceNameIsNull_ = false;}
   else{serviceNameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- serviceNameNode , address : " << serviceNameNode << std::endl;
   #endif
   if(serviceNameNode)
   {
      if(serviceNameNode->Attribute("href") || serviceNameNode->Attribute("id"))
      {
          if(serviceNameNode->Attribute("id"))
          {
             serviceNameIDRef_ = serviceNameNode->Attribute("id");
             serviceName_ = boost::shared_ptr<XsdTypeNormalizedString>(new XsdTypeNormalizedString(serviceNameNode));
             serviceName_->setID(serviceNameIDRef_);
             IDManager::instance().setID(serviceNameIDRef_,serviceName_);
          }
          else if(serviceNameNode->Attribute("href")) { serviceNameIDRef_ = serviceNameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { serviceName_ = boost::shared_ptr<XsdTypeNormalizedString>(new XsdTypeNormalizedString(serviceNameNode));}
   }

   //statusNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* statusNode = xmlNode->FirstChildElement("status");

   if(statusNode){statusIsNull_ = false;}
   else{statusIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- statusNode , address : " << statusNode << std::endl;
   #endif
   if(statusNode)
   {
      if(statusNode->Attribute("href") || statusNode->Attribute("id"))
      {
          if(statusNode->Attribute("id"))
          {
             statusIDRef_ = statusNode->Attribute("id");
             status_ = boost::shared_ptr<ServiceStatus>(new ServiceStatus(statusNode));
             status_->setID(statusIDRef_);
             IDManager::instance().setID(statusIDRef_,status_);
          }
          else if(statusNode->Attribute("href")) { statusIDRef_ = statusNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { status_ = boost::shared_ptr<ServiceStatus>(new ServiceStatus(statusNode));}
   }

   //processingStatusNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* processingStatusNode = xmlNode->FirstChildElement("processingStatus");

   if(processingStatusNode){processingStatusIsNull_ = false;}
   else{processingStatusIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- processingStatusNode , address : " << processingStatusNode << std::endl;
   #endif
   if(processingStatusNode)
   {
      if(processingStatusNode->Attribute("href") || processingStatusNode->Attribute("id"))
      {
          if(processingStatusNode->Attribute("id"))
          {
             processingStatusIDRef_ = processingStatusNode->Attribute("id");
             processingStatus_ = boost::shared_ptr<ServiceProcessingStatus>(new ServiceProcessingStatus(processingStatusNode));
             processingStatus_->setID(processingStatusIDRef_);
             IDManager::instance().setID(processingStatusIDRef_,processingStatus_);
          }
          else if(processingStatusNode->Attribute("href")) { processingStatusIDRef_ = processingStatusNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { processingStatus_ = boost::shared_ptr<ServiceProcessingStatus>(new ServiceProcessingStatus(processingStatusNode));}
   }

   //advisoryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* advisoryNode = xmlNode->FirstChildElement("advisory");

   if(advisoryNode){advisoryIsNull_ = false;}
   else{advisoryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- advisoryNode , address : " << advisoryNode << std::endl;
   #endif
   if(advisoryNode)
   {
      if(advisoryNode->Attribute("href") || advisoryNode->Attribute("id"))
      {
          if(advisoryNode->Attribute("id"))
          {
             advisoryIDRef_ = advisoryNode->Attribute("id");
             advisory_ = boost::shared_ptr<ServiceAdvisory>(new ServiceAdvisory(advisoryNode));
             advisory_->setID(advisoryIDRef_);
             IDManager::instance().setID(advisoryIDRef_,advisory_);
          }
          else if(advisoryNode->Attribute("href")) { advisoryIDRef_ = advisoryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { advisory_ = boost::shared_ptr<ServiceAdvisory>(new ServiceAdvisory(advisoryNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeNormalizedString> ServiceNotification::getServiceName()
{
   if(!this->serviceNameIsNull_){
        if(serviceNameIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNormalizedString>(IDManager::instance().getID(serviceNameIDRef_));
        }else{
             return this->serviceName_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNormalizedString>();
   }
}
boost::shared_ptr<ServiceStatus> ServiceNotification::getStatus()
{
   if(!this->statusIsNull_){
        if(statusIDRef_ != ""){
             return boost::shared_static_cast<ServiceStatus>(IDManager::instance().getID(statusIDRef_));
        }else{
             return this->status_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ServiceStatus>();
   }
}
boost::shared_ptr<ServiceProcessingStatus> ServiceNotification::getProcessingStatus()
{
   if(!this->processingStatusIsNull_){
        if(processingStatusIDRef_ != ""){
             return boost::shared_static_cast<ServiceProcessingStatus>(IDManager::instance().getID(processingStatusIDRef_));
        }else{
             return this->processingStatus_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ServiceProcessingStatus>();
   }
}
boost::shared_ptr<ServiceAdvisory> ServiceNotification::getAdvisory()
{
   if(!this->advisoryIsNull_){
        if(advisoryIDRef_ != ""){
             return boost::shared_static_cast<ServiceAdvisory>(IDManager::instance().getID(advisoryIDRef_));
        }else{
             return this->advisory_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ServiceAdvisory>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-msg-5-4/ServiceNotification.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ServiceNotification : NotificationMessage
    {
        public ServiceNotification(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode serviceNameNode = xmlNode.SelectSingleNode("serviceName");
            
            if (serviceNameNode != null)
            {
                if (serviceNameNode.Attributes["href"] != null || serviceNameNode.Attributes["id"] != null) 
                {
                    if (serviceNameNode.Attributes["id"] != null) 
                    {
                        serviceNameIDRef_ = serviceNameNode.Attributes["id"].Value;
                        XsdTypeNormalizedString ob = new XsdTypeNormalizedString(serviceNameNode);
                        IDManager.SetID(serviceNameIDRef_, ob);
                    }
                    else if (serviceNameNode.Attributes["href"] != null)
                    {
                        serviceNameIDRef_ = serviceNameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        serviceName_ = new XsdTypeNormalizedString(serviceNameNode);
                    }
                }
                else
                {
                    serviceName_ = new XsdTypeNormalizedString(serviceNameNode);
                }
            }
            
        
            XmlNode statusNode = xmlNode.SelectSingleNode("status");
            
            if (statusNode != null)
            {
                if (statusNode.Attributes["href"] != null || statusNode.Attributes["id"] != null) 
                {
                    if (statusNode.Attributes["id"] != null) 
                    {
                        statusIDRef_ = statusNode.Attributes["id"].Value;
                        ServiceStatus ob = new ServiceStatus(statusNode);
                        IDManager.SetID(statusIDRef_, ob);
                    }
                    else if (statusNode.Attributes["href"] != null)
                    {
                        statusIDRef_ = statusNode.Attributes["href"].Value;
                    }
                    else
                    {
                        status_ = new ServiceStatus(statusNode);
                    }
                }
                else
                {
                    status_ = new ServiceStatus(statusNode);
                }
            }
            
        
            XmlNode processingStatusNode = xmlNode.SelectSingleNode("processingStatus");
            
            if (processingStatusNode != null)
            {
                if (processingStatusNode.Attributes["href"] != null || processingStatusNode.Attributes["id"] != null) 
                {
                    if (processingStatusNode.Attributes["id"] != null) 
                    {
                        processingStatusIDRef_ = processingStatusNode.Attributes["id"].Value;
                        ServiceProcessingStatus ob = new ServiceProcessingStatus(processingStatusNode);
                        IDManager.SetID(processingStatusIDRef_, ob);
                    }
                    else if (processingStatusNode.Attributes["href"] != null)
                    {
                        processingStatusIDRef_ = processingStatusNode.Attributes["href"].Value;
                    }
                    else
                    {
                        processingStatus_ = new ServiceProcessingStatus(processingStatusNode);
                    }
                }
                else
                {
                    processingStatus_ = new ServiceProcessingStatus(processingStatusNode);
                }
            }
            
        
            XmlNode advisoryNode = xmlNode.SelectSingleNode("advisory");
            
            if (advisoryNode != null)
            {
                if (advisoryNode.Attributes["href"] != null || advisoryNode.Attributes["id"] != null) 
                {
                    if (advisoryNode.Attributes["id"] != null) 
                    {
                        advisoryIDRef_ = advisoryNode.Attributes["id"].Value;
                        ServiceAdvisory ob = new ServiceAdvisory(advisoryNode);
                        IDManager.SetID(advisoryIDRef_, ob);
                    }
                    else if (advisoryNode.Attributes["href"] != null)
                    {
                        advisoryIDRef_ = advisoryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        advisory_ = new ServiceAdvisory(advisoryNode);
                    }
                }
                else
                {
                    advisory_ = new ServiceAdvisory(advisoryNode);
                }
            }
            
        
        }
        
    
        #region ServiceName_
        private XsdTypeNormalizedString serviceName_;
        public XsdTypeNormalizedString ServiceName_
        {
            get
            {
                if (this.serviceName_ != null)
                {
                    return this.serviceName_; 
                }
                else if (this.serviceNameIDRef_ != null)
                {
                    serviceName_ = IDManager.getID(serviceNameIDRef_) as XsdTypeNormalizedString;
                    return this.serviceName_; 
                }
                else
                {
                      return this.serviceName_; 
                }
            }
            set
            {
                if (this.serviceName_ != value)
                {
                    this.serviceName_ = value;
                }
            }
        }
        #endregion
        
        public string serviceNameIDRef_ { get; set; }
        #region Status_
        private ServiceStatus status_;
        public ServiceStatus Status_
        {
            get
            {
                if (this.status_ != null)
                {
                    return this.status_; 
                }
                else if (this.statusIDRef_ != null)
                {
                    status_ = IDManager.getID(statusIDRef_) as ServiceStatus;
                    return this.status_; 
                }
                else
                {
                      return this.status_; 
                }
            }
            set
            {
                if (this.status_ != value)
                {
                    this.status_ = value;
                }
            }
        }
        #endregion
        
        public string statusIDRef_ { get; set; }
        #region ProcessingStatus_
        private ServiceProcessingStatus processingStatus_;
        public ServiceProcessingStatus ProcessingStatus_
        {
            get
            {
                if (this.processingStatus_ != null)
                {
                    return this.processingStatus_; 
                }
                else if (this.processingStatusIDRef_ != null)
                {
                    processingStatus_ = IDManager.getID(processingStatusIDRef_) as ServiceProcessingStatus;
                    return this.processingStatus_; 
                }
                else
                {
                      return this.processingStatus_; 
                }
            }
            set
            {
                if (this.processingStatus_ != value)
                {
                    this.processingStatus_ = value;
                }
            }
        }
        #endregion
        
        public string processingStatusIDRef_ { get; set; }
        #region Advisory_
        private ServiceAdvisory advisory_;
        public ServiceAdvisory Advisory_
        {
            get
            {
                if (this.advisory_ != null)
                {
                    return this.advisory_; 
                }
                else if (this.advisoryIDRef_ != null)
                {
                    advisory_ = IDManager.getID(advisoryIDRef_) as ServiceAdvisory;
                    return this.advisory_; 
                }
                else
                {
                      return this.advisory_; 
                }
            }
            set
            {
                if (this.advisory_ != value)
                {
                    this.advisory_ = value;
                }
            }
        }
        #endregion
        
        public string advisoryIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

